addons/account/static/src/components/document_file_uploader/document_file_uploader.js

```diff
@@ -0,0 +1,77 @@
+import { useService } from "@web/core/utils/hooks";
+import { FileUploader } from "@web/views/fields/file_handler";
+import { standardWidgetProps } from "@web/views/widgets/standard_widget_props";
+
+import { Component, markup } from "@odoo/owl";
+
+export class DocumentFileUploader extends Component {
+    static template = "account.DocumentFileUploader";
+    static components = {
+        FileUploader,
+    };
+    static props = {
+        ...standardWidgetProps,
+        record: { type: Object, optional: true },
+        divClass: { type: String, optional: true },
+        slots: { type: Object, optional: true },
+        resModel: { type: String, optional: true },
+    };
+
+    setup() {
+        this.orm = useService("orm");
+        this.action = useService("action");
+        this.notification = useService("notification");
+        this.attachmentIdsToProcess = [];
+        this.extraContext = this.getExtraContext();
+    }
+
+    // To pass extra context while creating record
+    getExtraContext() {
+        return {};
+    }
+
+    async onFileUploaded(file) {
+        const att_data = {
+            name: file.name,
+            mimetype: file.type,
+            datas: file.data,
+        };
+        const [att_id] = await this.orm.create("ir.attachment", [att_data], {
+            context: { ...this.extraContext, ...this.env.searchModel.context },
+        });
+        this.attachmentIdsToProcess.push(att_id);
```

Comment:
All internal users are allowed to create attachments, but this doesn't mean that they are allowed to create invoice/SO/..., even if they have read rights on the model.
I believe this should be improved in the future (let's keep it like that for now, it was already like that)